src/components/price-color-legend.tsx

```diff
@@ -14,40 +15,28 @@ const TOP_LABEL_HEIGHT = 14;
 const COLOR_HEIGHT = 12;
 const BOTTOM_LABEL_HEIGHT = 16;
 
-const LegendBox = ({
-  children,
-  sx,
-}: {
-  children: React.ReactNode;
-  sx?: BoxProps["sx"];
-}) => {
-  return (
-    <Box
-      sx={{
-        zIndex: 13,
-        bgcolor: "rgba(245, 245, 245, 0.8)",
-        borderRadius: 1,
-        height: "fit-content",
-        px: 4,
-        py: 2,
-        ...sx,
-      }}
-    >
-      {children}
-    </Box>
-  );
-};
+const LegendBox = styled(Box)({
+  zIndex: 13,
+  bgcolor: "rgba(245, 245, 245, 0.8)",
+  borderRadius: 1,
+  height: "fit-content",
+  px: 4,
+  py: 2,
+});
+
 export const MapPriceColorLegend = ({
   stats,
+  id,
 }: {
   stats: [number | undefined, number | undefined, number | undefined];
+  id?: string;
```

Comment:
Nit: Maybe `id` should be non-nullable to make sure the legend is always screenshotted?